src/main/java/com/artipie/api/artifactory/CreateRepoSlice.java

```diff
@@ -0,0 +1,124 @@
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import javax.json.JsonObject;
+import org.reactivestreams.Publisher;
+
+/**
+ * Artifactory create repo API slice, it accepts json and create new docker repository by
+ * creating corresponding YAML configuration.
+ * @since 0.9
+ */
+public final class CreateRepoSlice implements Slice {
+
+    /**
+     * Artipie settings.
+     */
+    private final Settings settings;
+
+    /**
+     * Ctor.
+     * @param settings Artipie settings
+     */
+    public CreateRepoSlice(final Settings settings) {
+        this.settings = settings;
+    }
+
+    @Override
+    public Response response(
+        final String line,
+        final Iterable<Map.Entry<String, String>> headers,
+        final Publisher<ByteBuffer> body
+    ) {
+        // @checkstyle ReturnCountCheck (20 lines)
+        return new AsyncResponse(
+            Single.just(body).to(ContentAs.JSON).flatMap(
+                json -> Single.fromFuture(
+                    valid(json).map(
+                        name -> {
+                            try {
+                                return this.settings.storage().exists(
+                                    new Key.From(String.format("%s.yaml", name))
+                                ).thenApply(
+                                    exists -> {
+                                        final Response res;
+                                        if (exists) {
+                                            res = new RsWithStatus(RsStatus.BAD_REQUEST);
+                                        } else {
+                                            res = new RsWithStatus(RsStatus.OK);
+                                        }
+                                        return res;
+                                    }
+                                );
+                            } catch (final IOException ex) {
+                                Logger.error(this, ex.toString());
+                                return CompletableFuture.completedFuture(
+                                    new RsWithStatus(RsStatus.INTERNAL_ERROR)
+                                );
+                            }
+                        }
+                    ).orElse(
+                        CompletableFuture.completedFuture(new RsWithStatus(RsStatus.BAD_REQUEST))
+                    )
+                )
+            )
+        );
+    }
+
+    /**
+     * Checks if json is valid (contains new repo key and supported setting) and
+     * return new repo name.
+     * @param json Json to read repo name from
+     * @return True if json is correct
+     */
+    private static Optional<String> valid(final JsonObject json) {
+        final Optional<String> res;
+        final String key = json.getString("key", "");
+        if (!key.isEmpty() && "local".equals(json.getString("rclass", ""))
```

Comment:
@olenagerasimova we may need to support different parameters later, e.g. remote or virtual `rclass` and non docker repositories

Reply:
@g4s8 hm, do not see any problems: we introduce class for these parameters and this method will return optional of this new class instead of just string. Or do you think I should do it now?